### ISSUE TITLE
Accumulating into data slices doesn't work with MPI

### ISSUE THREAD

**Issue by mikeando** [opesci/devito#1007]
The following small program errors when run with 2 MPI ranks. The error is:

~~~
ValueError: could not broadcast input array from shape (16,31) into shape (0,30,31)
ValueError: could not broadcast input array from shape (16,31) into shape (16,30,31)
~~~

~~~
from devito import Function, Grid, TimeFunction, Eq, Operator, configuration
from mpi4py import MPI
configuration['mpi'] = True

grid = Grid(shape=(31,31,31), comm=MPI.COMM_WORLD)
f = Function(name="f", grid=grid)
f.data[:] = 1

#OK
print("Adding f[1,:,:] to f[0,:,:]")
f.data[1,:,:] = f.data[1,:,:] + f.data[0,:,:]
f.data[1,:,:] += f.data[0,:,:]

#Neither of these is OK
print("Adding f[:,1,:] to f[:,0,:]")
f.data[:,1,:] = f.data[:,1,:] + f.data[:,0,:]
f.data[:,1,:] += f.data[:,0,:]
~~~

**Comment by rhodrin** [comment]
Thanks for the report @mikeando. Looking into this at the moment, should have a fix at some point today.

**Comment by rhodrin** [comment]
This should be fixed on https://github.com/opesci/devito/tree/misc-data-fixes.

**Comment by FabioLuporini** [comment]
closing as fixed by #1008
